floss/main.py

```diff
@@ -536,29 +521,74 @@ def main(argv=None) -> int:
 
     results = ResultDocument(metadata=Metadata(file_path=str(sample), min_length=args.min_length), analysis=analysis)
 
-    time0 = time()
-    interim = time0
     sample_size = sample.stat().st_size
+    if sample_size > sys.maxsize:
+        logger.warning("file is very large, strings listings may be truncated.")
 
+    # always extract static strings, it's fast and we use them for language identification
+    # can throw away result later if not desired in output
+    time0 = time()
+    interim = time0
     static_strings = get_static_strings(sample, args.min_length)
+    static_runtime = get_runtime_diff(interim)
+    interim = time()
+
+    language = identify_language(sample, static_strings)
 
-    language = identify_language(sample=sample, static_strings=static_strings)
+    # set language configurations
+    if language == language.GO:
+        results.metadata.language = language.GO.value
+
+        prompt = input("Do you want to enable string deobfuscation? (this could take a long time) [y/N] ")
+
+        if prompt == "y" or prompt == "Y":
+            logger.info("enabling string deobfuscation...")
+            analysis.enable_stack_strings = True
+            analysis.enable_tight_strings = True
+            analysis.enable_decoded_strings = True
+
+        else:
+            logger.info("disabling string deobfuscation...")
+            analysis.enable_stack_strings = False
+            analysis.enable_tight_strings = False
+            analysis.enable_decoded_strings = False
```

Comment:
What if users want to just set one type?
I think we should only ask here if neither `--no` nor `--only` are provided on the command line.